core/src/main/java/org/pentaho/reporting/platform/plugin/output/util/ExportReportUtils.java

```diff
@@ -0,0 +1,205 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.reporting.platform.plugin.output.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.pentaho.metadata.query.model.Query;
+import org.pentaho.pms.core.exception.PentahoMetadataException;
+import org.pentaho.reporting.engine.classic.core.Element;
+import org.pentaho.reporting.engine.classic.core.MasterReport;
+import org.pentaho.reporting.engine.classic.core.ReportDataFactoryException;
+import org.pentaho.reporting.engine.classic.core.ReportHeader;
+import org.pentaho.reporting.engine.classic.core.elementfactory.LabelElementFactory;
+import org.pentaho.reporting.engine.classic.core.parameters.DefaultParameterContext;
+import org.pentaho.reporting.engine.classic.core.parameters.ParameterDefinitionEntry;
+import org.pentaho.reporting.libraries.formula.EvaluationException;
+import org.pentaho.reporting.libraries.formula.parser.ParseException;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public interface ExportReportUtils {
```

Comment:
I don't understand why this would be an interface. And have classes such as `SimpleReportingComponent` and `SimpleReportingAction` implement it just to gain a couple of static methods? The inherited instance methods are like utilities as well... No need to publicly expose the API of `ExportReportUtils` on these classes just for what appears to be gaining some syntactic sugar/convenience. I think consuming the utility via composition would be more adequate. You can even have it not have any public static methods.

Just saw the other related comments. I don't think the justification presented in https://github.com/pentaho/pentaho-platform-plugin-reporting/pull/899#discussion_r2136964765 applies to this use case.